app/models/mood.rb

```diff
@@ -0,0 +1,19 @@
+class Mood < ApplicationRecord  
+  validates :user_id, presence: true, numericality: true
+  validates :mood, presence: true, numericality: { greater_than_or_equal_to: 1, less_than_or_equal_to: 5 } 
+  validates :date, presence: true
+
+  before_validation :set_date, on: :create
+
+  def self.user_moods(user_id)
+    where(user_id: user_id)
+  end
+
+  def self.average_mood(user_id)
+    avg = where(user_id: user_id).average(:mood).to_f.round(1)
+  end
+
+  def set_date
+    self.date ||= Time.zone.now.strftime("%Y-%m-%d")
```

Comment:
Interesting, I was figuring we could just use the 'created at' column that gets added to each record on creation as our way of keeping track of the dates that a user input their mood, but this works too and makes the info a little more human readable without having to convert/change anything later!